src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison/PrisonerSearchApiClient.kt

```diff
@@ -11,13 +11,13 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.mode
 @Service
 class PrisonerSearchApiClient(@Qualifier("oauthPrisonerSearchClient") val prisonerSearchApiWebClient: WebClient) {
 
-  fun searchPrisonersByBookingIds(bookingIds: List<Long>): List<PrisonerSearchPrisoner> {
+  fun searchPrisonersByBookingIds(bookingIds: Collection<Long>): List<PrisonerSearchPrisoner> {
     if (bookingIds.isEmpty()) return emptyList()
     return prisonerSearchApiWebClient
       .post()
       .uri("/prisoner-search/booking-ids")
       .accept(MediaType.APPLICATION_JSON)
-      .bodyValue(PrisonerSearchByBookingIdsRequest(bookingIds))
+      .bodyValue(PrisonerSearchByBookingIdsRequest(bookingIds.toList()))
```

Comment:
Map doesn't implement the [Collection](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Collection.html) interface, If it did then it would probably be a `Collection<Entry<K, V>>` which wouldn't compile - you'd need to explicitly convert it into a list of longs as that's what `PrisonerSearchByBookingIdsRequest` accepts.

Historically its been seen to be good practice accept the most abstract type that fits the use case - it makes methods reusable - by converting the param type to a Collection, it now works with dozens of alternative implementations.